I24 unit tests failing

Currently, on main, the i24 tests are failing with:
```
FAILED tests/beamlines/unit_tests/test_i24.py::test_device_creation - AssertionError: assert 'BL23I-MO-VGON-01:' == 'BL24I-MO-VGON-01:'
  
  - BL24I-MO-VGON-01:
  ?    ^
  + BL23I-MO-VGON-01:
  ?    ^
```

## Acceptance Criteria
* Tests all pass

--- a/tests/beamlines/unit_tests/test_i24.py
+++ b/tests/beamlines/unit_tests/test_i24.py
@@ -1,11 +1,15 @@
-import sys
+import os
 from unittest.mock import patch
 
 from dodal.devices.i24.i24_vgonio import VGonio
 
-with patch.dict("os.environ", {"BEAMLINE": "i24"}, clear=True):
+with patch.dict(os.environ, {"BEAMLINE": "i24"}, clear=True):
     from dodal.beamlines import beamline_utils, i24
     from dodal.utils import make_all_devices
+
+
+def setup_module():
+    beamline_utils.set_beamline("i24")
 
 
 def test_device_creation():
@@ -21,7 +25,5 @@ def test_device_creation():
 
 
 def teardown_module():
-    beamline_utils.BL = "i03"
-    for module in list(sys.modules):
-        if module.endswith("beamline_utils"):
-            del sys.modules[module]
+    beamline_utils.set_beamline("i03")
+    beamline_utils.clear_devices()

